### PR TITLE
aws-c-mqtt: add version 0.7.9 (re-open)

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **aws-c-mqtt/0.7.9**

I reopen PR of https://github.com/conan-io/conan-center-index/pull/8092.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
@XuhuiSun95
I'm sorry the old PR is closed automatically by my other PR.
I reoopened it.

**Comment by XuhuiSun95** [comment]
@toge Seems like we still have the same issue happening here and my local `aws-cpp-sdk` build.

Is it possible @MartinDelille can take a look at the error? I think you created most of the aws related receipts, maybe you can give us some insight on how to fix the issue?

**Comment by toge** [comment]
I'm looking into it, but can't find a solution.
 
In my local environment, I can generate the package without any problem, so it is taking me a long time to investigate.

I am sorry for the inconvenience.

**Comment by MartinDelille** [comment]
I may work because you have the recipes in your cache and/or you are using the `--build missing` parameter instead of `--build aws-c-mqtt` when performing `conan test ...`

**Comment by toge** [comment]
@MartinDelille 
Thank you for your information.
I try to reproduce CI error following commands but test program succeeded.

```
conan remove -f aws-c-http/\*
conan remove -f aws-c-mqtt/\*
conan create . 0.7.9@ --build aws-c-mqtt --build aws-c-http
conan test test_package aws-c-mqtt/0.7.9@
```

Are there something wrong to reproduce?

**Comment by XuhuiSun95** [comment]
> @XuhuiSun95 @MartinDelille Now All CI checks have passed. Please tell me if there are still some problems to solve.

LGTM, thanks for your work!

How was the pipeline fixed, I didn't see many dependency updates since then. Was that only because we run the pipeline too fast since the code reverted, and the dependency packages have not yet been rebuilt by that time?

**Comment by toge** [comment]
@XuhuiSun95 
I don't know the reason in detail, but I suspect the same thing happened as you wrote.

File: recipes/aws-c-mqtt/all/conanfile.py [hunk 1]
```diff
@@ -69,10 +69,13 @@ def package(self):
     def package_info(self):
         self.cpp_info.filenames["cmake_find_package"] = "aws-c-mqtt"
         self.cpp_info.filenames["cmake_find_package_multi"] = "aws-c-mqtt"
+        self.cpp_info.set_property("cmake_file_name", "aws-c-mqtt")
         self.cpp_info.names["cmake_find_package"] = "AWS"
         self.cpp_info.names["cmake_find_package_multi"] = "AWS"
+        self.cpp_info.set_property("cmake_target_name", "AWS")
```
Comment (ericLemanissier) [inline recipes/aws-c-mqtt/all/conanfile.py:75]:
```suggestion
```

File: recipes/aws-c-mqtt/all/conanfile.py [hunk 2]
```diff
@@ -69,10 +69,13 @@ def package(self):
...
         self.cpp_info.components["aws-c-mqtt-lib"].names["cmake_find_package"] = "aws-c-mqtt"
         self.cpp_info.components["aws-c-mqtt-lib"].names["cmake_find_package_multi"] = "aws-c-mqtt"
+        self.cpp_info.components["aws-c-mqtt-lib"].set_property("cmake_target_name", "aws-c-mqtt")
```
Comment (ericLemanissier) [inline recipes/aws-c-mqtt/all/conanfile.py:78]:
```suggestion
        self.cpp_info.components["aws-c-mqtt-lib"].set_property("cmake_target_name", "AWS::aws-c-mqtt")
```

**Comment by ericLemanissier** [comment]
well, actually, conan considers cmake target name as absolute starting with conan 1.43.0, which is not yet deployed on CCI. It's a bit too early for this change, sorry :/

**Comment by toge** [comment]
@ericLemanissier 
OK.
Your comment always help me because I have no time to follow current trend of conan.

I will revert latest commit later.
When CI update to 1.43.0, I will follow your points!
Thank you!

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**